cng/tls1prf.go

```diff
@@ -24,26 +24,29 @@ func loadTLS1PRF(id string) (bcrypt.ALG_HANDLE, error) {
 	return h.(bcrypt.ALG_HANDLE), nil
 }
 
-func TLS1PRF(secret, label, seed []byte, keyLen int, h func() hash.Hash) ([]byte, error) {
+// TLS1PRF implements the TLS 1.0/1.1 pseudo-random function if h is nil or crypto.MD5SHA1,
+// else it implements the TLS 1.2 pseudo-random function.
+// The pseudo-random number will be written to result and will be of length len(result).
+func TLS1PRF(result, secret, label, seed []byte, h func() hash.Hash) error {
 	// TLS 1.0/1.1 PRF uses MD5SHA1.
 	algID := bcrypt.TLS1_1_KDF_ALGORITHM
 	var hashID string
 	if h != nil {
 		// If h is specified, assume the caller wants to use TLS 1.2 PRF.
 		// TLS 1.0/1.1 PRF doesn't allow specifying the hash function.
 		if hashID = hashToID(h()); hashID == "" {
```

Comment:
I may be wrong here but does this not also need to check if `hashID == "crypto.MD5SHA1"`?

Reply:
Huh, I copy pasted the comment from the OpenSSL backend, and forgot that CNG doesn't support automatically generates a TLS 1.0/1.1 number when using `crypto.MD5SHA1`. I wasn't planning to special case it, as in the Go standard library we use `nil` to identify TLS 1.0/1.1. But, now that I'm here, I'll implement the `crypto.MD5SHA1` special case, the more similar that the CNG and OpenSSL backends are, the better.

Reply:
Forget my previous comment, I just remembered why I didn't special case `crypto.MD5SHA1`: CNG doesn't support it, so there is no way for `h` to contain a valid md5sha1 hasher. I'll update the comment.